### ISSUE TITLE
is there a way customize pivot model?

### ISSUE THREAD

**Issue by jamelwa** [spatie/laravel-permission#966]
hi there, thanks for the great package you provide for us. i have one issue, i want to customize pivot model based on `model has roles` table.

here's what i'm trying to achieve:
- i have `document` table.
- if `user` has many to many relationship with `roles`. 
- i'm wondering.. can i use `hasManyThrough` relationship?

so the final result will be like this:
- `role` has many `documents` through `user`.

tldr: i only want to write my pivot model between `user` and `role`

**Comment by drbyte** [comment]
What is your use-case for this? 
By saying "role has many documents through user", does that mean you're looking for a means of getting "when I specify a role, I can use the user relationship to find all documents associated with all users assigned to that role"?

I'm trying to understand whether you're asking a question about this package specifically, vs a general question about using existing Laravel relationships to do additional lookups?

**Comment by jamelwa** [comment]
yes, i want to retrieve all the documents associated with all user with specified role. but i think i can figure it out myself.

so sorry if it's out of topic,
i just want to ask how do i create basic pivot model between role model and user model, based on this package's migration table?

**Comment by freekmurze** [comment]
Hi @jamelwa,

that's a question not about this package, but on general Laravel development. Could you post your question on public fora like the Laracasts forum, Laravel.io or Stack Overlow?

Good luck!
